documentation/assets/Example: training an external model.md

```diff
@@ -0,0 +1,753 @@
+        super(CustomSegLoss, self).__init__()
+
+        self.bce_weight = bce_weight
+        self.iou_weight = iou_weight
+
+        self.bce_loss = nn.BCELoss()
+        self.iou_func = JaccardIndex(num_classes=2)
+
+    def forward(self, preds, target):
+
+        bce_loss = self.bce_loss(preds, target)
+        iou_loss = 1.0 - self.iou_func(torch.gt(preds, 0.5).long(), torch.gt(target, 0.5).long())
+
+        return self.bce_weight*bce_loss + self.iou_weight*iou_loss
+```
+
+Notice that here the BCE loss term is obtained simply by using PyTorch's `BCELoss()`. To compute the IoU score, or
+the Jaccard Index, we use torchmetric's `JaccardIndex`. To compute the IoU, the pixel values in both images should
+be binary, i.e., 0's and 1's. Since in the model's forward function a sigmoid function is already applied to the
+output, we only need to binarize both the predictions and the target tensors with a threshold of 0.5 (remember, we are 
+using **soft** masks). When measuring segmentation performance, higher IoU is better. Since IoU score is a number 
+in [0, 1], we simply compute `IoU loss = 1 - IoU` to make it a valid loss function for gradient-descent optimization.
+
+The overall loss is a weighted sum of the BCE and the IoU loss terms, with weights `bce_weight` and `iou_weight`,
+respectively. 
+
+With just a few lines of code we have defined our own custom loss function. Although here we could have used a similar 
+loss function provided by SuperGradients, in many other cases a more complex and specific loss function is desired. 
+This example serves to show how we may define any loss function to fit our needs and seamlessly integrate it with the
+SuperGradients training pipeline.
+
+## 4. Custom IoU metric
+
+Since our measure of success is the IoU, we would like to tell SuperGradients to log and track this metric during
+training. This metric would also be used to determine the best model at the end of every epoch for checkpointing.
+We note that SuperGradients provides many built-in metrics, including variants of the IoU. However, in this example 
+we aim to show the ease at which we can incorporate external metrics into our pipeline. SuperGradients supports any 
+metric of type `torchmetrics.Metric`.
+
+The metric we will use is torchmetrics' `JaccardIndex`, which was also used to implement our loss function. However,
+recall that we use soft masks. 
+[JaccardIndex](https://torchmetrics.readthedocs.io/en/stable/classification/jaccard_index.html) requires the target
+tensor's elements to be integers. Therefore, we will need to modify it a bit. For more details about implementing a 
+custom metric using torchmetrics, see [here](https://torchmetrics.readthedocs.io/en/stable/pages/implement.html).
+
+```
+class SoftIoU(JaccardIndex):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def update(self, preds: torch.Tensor, target: torch.Tensor):
+        target = torch.gt(target, 0.5).long()
```

Comment:
This will take the mean of the ious for both background and the target 
(therefore the results are so promising so fast).

For binary segmentation it is more common to take the you of target only.

Reply:
Should be something like this:

class SoftIoU(JaccardIndex):
     def __init__(self, **kwargs):
         super().__init__(reduction="none", **kwargs)
     def update(self, preds: torch.Tensor, target: torch.Tensor):
         target = torch.gt(target, 0.5).long()
         super().update(preds, target)
     def compute(self):
          return super().compute()[1]

Reply:
nice one! 
changed according to your suggestion